storage/jellyfish-merkle/src/node_type/mod.rs

```diff
@@ -724,7 +724,7 @@ where
     /// Returns `NodeType`
     pub fn node_type(&self) -> NodeType {
         match self {
-            Self::Null => unreachable!(),
+            Self::Null => unreachable!("Null won't be a child of an internal node."),
```

Comment:
I think a comment should explain why this method would not be called by Null node.